### DISCUSSION TITLE
k3s AWS Credential support

### DISCUSSION THREAD

**Discussion by rajasec** [k3s-io/k3s#10555] [Q&A]
**Environmental Info:**
K3s Version: 

k3s version v1.29.6+k3s2 (b4b156d9)
Node(s) CPU architecture, OS, and Version: 

Linux 5.15 x86_64 x86_64 GNU/Linux
Cluster Configuration:k3s version v1.29.6+k3s2 (b4b156d9)
**Describe the bug:**

We have k3s deployment which will be installed in hospital environment, we need to leverage the credential feature provided by k8s upstream release. Need clarity whether k3s support this feature so that we dont need to manage the static credentials by passing this creds part of kubelet. In-tree provided cred provider has been disabled from k8s. Would like to understand whether cloud provider plugins are supported ( Eg: ECR Cred provider)

**Steps To Reproduce:**

- Installed K3s: 

**Expected behavior:**

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
Credential provider plugins are standalone binaries that are called by the kubelet. Similarly, the out-of-tree cloud controllers are deployed to the cluster as helm charts or flat manifests. Nothing we've removed from K3s would prevent you from using either of these.

**Comment by rajasec** [comment]
@brandond 
Thank you for the quick response. We will be using out of tree cloud controllers ( moreover in-tree plugins has been deprecated if I'm not wrong from 1.28 onwards).
This helps to proceed to secure our hospital on-prem environment with dynamic creds.
